### PR TITLE
✨ [JDBC-V2] Data type formatting when constructing when using Prepared Statement 

### PR DESCRIPTION
**Pull request by chernser**
…issue with Date/Time data
This pull request introduces new formatting utilities for date and time types in the ClickHouse Java client and adds a configurable object encoding strategy for JDBC `PreparedStatement`. The main changes include new helper methods for formatting Java date/time objects for ClickHouse SQL, comprehensive integration tests for these methods, and a new driver property allowing users to choose between CAST-based and direct object conversion strategies for parameter encoding.

**Date/Time Formatting Utilities:**

* Added static methods to `DataTypeUtils` for formatting `Date`, `Time`, `Timestamp`, `Instant`, `OffsetDateTime`, and `ZonedDateTime` into ClickHouse-compatible SQL strings. These methods ensure correct formatting for use with ClickHouse functions like `parseDateTimeBestEffort`.
* Introduced extensive integration tests in `DataTypeTests` to verify the new formatting methods produce correct and ClickHouse-parseable strings, including edge cases with nanosecond precision and time zone normalization.

**JDBC PreparedStatement Object Encoding Strategy:**

* Added a new driver property `jdbc_object_encode_strategy` (`OBJECT_ENCODE_STRATEGY`) to configure how objects are encoded in `PreparedStatement`. Supported values are `CAST` (uses SQL CAST for type conversion) and `CONVERSION` (direct conversion).
* Refactored `PreparedStatementImpl` to use a pluggable encoding strategy (`ObjectEncodeStrategy` interface) with two implementations: `CastEncodeStrategy` and `ObjectConversionEncodeStrategy`. The strategy is selected based on the driver property at statement creation. [[1]](diffhunk://#diff-b3fdfecf6eb5032594e47bfb9c6c4b28d54438e401263c6ff10bb068c407487fL1034-R1055) [[2]](diffhunk://#diff-b3fdfecf6eb5032594e47bfb9c6c4b28d54438e401263c6ff10bb068c407487fR1070-R1105) [[3]](diffhunk://#diff-8953c50a58dbb59ea6c90c09b7bb7f7260a8a5cc02b31b774f4599ac08099e9eL399-R411)
* Updated integration tests in `PreparedStatementTest` to explicitly set the encoding strategy to `CAST` via connection properties, ensuring type casting is tested according to the new configuration. [[1]](diffhunk://#diff-bd79e31286cff987e50b80a6dac3c02d10bb23cf5d4b78c54c10ea84441caa34L1411-R1413) [[2]](diffhunk://#diff-bd79e31286cff987e50b80a6dac3c02d10bb23cf5d4b78c54c10ea84441caa34L1451-R1455)

## Summary

Closes 
## Checklist
Delete items not relevant to your PR:
- [ ] Closes #<issue ref>
- [ ] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: jdbc-v2/src/main/java/com/clickhouse/jdbc/ConnectionImpl.java [hunk 1]
```diff
@@ -396,7 +396,19 @@ public PreparedStatement prepareStatement(String sql, int resultSetType, int res
                 return new WriterStatementImpl(this, sql, tableSchema, parsedStatement);
             }
         }
-        return new PreparedStatementImpl(this, sql, parsedStatement);
+        PreparedStatementImpl preparedStatement = new PreparedStatementImpl(this, sql, parsedStatement);
+        
+        // Set object encode strategy based on configuration
+        String strategyName = config.getDriverProperty(DriverProperties.OBJECT_ENCODE_STRATEGY.getKey(),
+                DriverProperties.OBJECT_ENCODE_STRATEGY.getDefaultValue());
+        if ("CAST".equalsIgnoreCase(strategyName)) {
```
[the remark this answers is not in the document]
Comment (chernser) [reply jdbc-v2/src/main/java/com/clickhouse/jdbc/ConnectionImpl.java:404]:
```java 
public boolean equalsIgnoreCase(String anotherString) {
        return (this == anotherString) ? true
                : (anotherString != null)
                && (anotherString.length() == length())
                && regionMatches(true, 0, anotherString, 0, length());
    }
```

seems there is a null check ;-)
